copter/source/docs/weathervaning.rst

```diff
@@ -0,0 +1,73 @@
+
+The way it works is the autopilot looks at the roll and/or pitch attitude needed to
+control the desired position. The basic algorithm is "yaw into the
+roll/pitch". If the aircraft needs hold roll to the right in order to hold
+position then it will turn in that direction on the assumption that
+the right roll is needed in order to hold against the wind (assuming nose into the wind). Similarly, for Side Into the Wind, it will use the pitch angle and yaw appropriately, trying to zero the pitch required to hold position.
+
+.. note:: by default, weathervaning does not use pitch, only roll. This is to prevent unwanted continuous yawing if the hover attitude has not been trimmed to hover, in place, with no wind. If this **has** been done, then pitch driven weathervaning can be enabled by setting :ref:`WVANE_OPTIONS<WVANE_OPTIONS>` bit 0 to "1". This will speed weathervaning if positioned with the wind blowing from behind (ie mostly being held in position with pitch, not roll). This does not affect Side Into Wind, it always uses pitch, hover trim attitude must be properly set for good operation.
+
+How quickly the aircraft yaws is determined by the :ref:`WVANE_GAIN<WVANE_GAIN>`
+parameter. It converts the lean angle into degs/sec of yaw. A good value to start with is 1 (1 degree roll = 1 deg/sec yaw). Higher values will make
+the aircraft turn into the roll more quickly. If the value is too high
+then you can get instability and oscillation in yaw.
+
+To cope with a small amount of trim in the aircraft there is an
+additional parameter :ref:`WVANE_ANG_MIN<WVANE_ANG_MIN>` which
+controls the minimum roll/pitch level before weathervaning will be
+used. This defaults to one degree. If you find your aircraft starts
+yawing even in no wind then you may need to raise this value.
+
+There are a number of additional parameters that can control when WeatherVaning is active (all are disabled by default):
+
+- :ref:`WVANE_HGT_MIN<WVANE_HGT_MIN>`: above this height weathervaning is permitted
+- :ref:`WVANE_SPD_MAX<WVANE_SPD_MAX>`: below this ground speed weathervaning is permitted
+- :ref:`WVANE_VELZ_MAX<WVANE_VELZ_MAX>`: maximum climb or descent speed at which the vehicle will still attempt to weathervane
+- :ref:`WVANE_TAKEOFF<WVANE_TAKEOFF>`: override weathervaning direction in auto takeoffs*
+- :ref:`WVANE_LAND<WVANE_LAND>`: overide weathervaning directions in auto landings*
+
+.. note:: to allow weathervaning only on takeoffs and landing during an AUTO mission or in GUIDED mode, set :ref:`WVANE_ENABLE<WVANE_ENABLE>` =0 but use the :ref:`WVANE_LAND<WVANE_LAND>` and/or :ref:`WVANE_TAKEOFF<WVANE_TAKEOFF>` parameters to enable the weathervaning control.
```

Comment:
Setting to 0 globally disables, you just have to set to some other invalid value. We should probably make that more clear in the code to (I have not tested if it this does actually work).

The typical use case in plane would be globally enabled but override to disabled for takeoff so you know the transition direction ahead of time.

Reply:
but enabling and overriding for takeoff and landing leaves it working on all other flight regimes....something I think most would not want...rather ONLY in landing....and there are no transitions in Copter

Reply:
#23054 addresses this...when merged will add to this doc